src/main/java/com/hivemq/extensions/prometheus/PrometheusExtensionMain.java

```diff
@@ -61,15 +64,22 @@ public void extensionStart(
                     ((e.getMessage() != null) ? ": " + e.getMessage() : ""));
             return;
         }
-
-        prometheusServer = new PrometheusServer(configuration, Services.metricRegistry());
-        prometheusServer.start();
+        try {
+            final PrometheusServer prometheusServer = new PrometheusServer(configuration, Services.metricRegistry());
+            prometheusServer.start();
+            prometheusServerRef.set(prometheusServer);
+        } catch (final Exception e) {
+            LOG.error("Error starting the Jetty Server for Prometheus Extension", e);
```

Comment:
We now catch all startup failures of the Jetty server here, log the original exception and prevent the extension startup properly.